src/schema/v2/artworksForUser/helpers.ts

```diff
@@ -33,7 +43,13 @@ export const getNewForYouRecs = async (
           }
         }
       `,
-  })()
+  }
+
+  const isAuthenticatedRequest = !!vortexGraphQLAuthenticatedLoader
+
+  const vortexResult = isAuthenticatedRequest
+    ? await vortexGraphQLAuthenticatedLoader(query)()
```

Comment:
So does this work? If a request contains the impersonate headers, it still counts as authenticated, but then this loader will fail
<img width="1004" alt="Screen Shot 2023-11-07 at 10 18 42 AM" src="https://github.com/artsy/metaphysics/assets/1457859/fea1ee91-5ed6-45dd-b68b-cc63ca1713e4">

Notice how it hits a `me` route. Maybe it's better to check `xImpersonateUserId` instead?

Reply:
makes sense!